instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/Instrumenter.java

```diff
@@ -121,19 +119,19 @@ public Context start(Context parentContext, REQUEST request) {
       spanBuilder.setStartTimestamp(startTimeExtractor.extract(request));
     }
 
-    AttributesBuilder attributesBuilder = Attributes.builder();
+    UnsafeAttributes attributesBuilder = new UnsafeAttributes();
     for (AttributesExtractor<? super REQUEST, ? super RESPONSE> extractor : extractors) {
       extractor.onStart(attributesBuilder, request);
     }
-    Attributes attributes = attributesBuilder.build();
+    Attributes attributes = attributesBuilder;
 
     Context context = parentContext;
 
     for (RequestListener requestListener : requestListeners) {
       context = requestListener.start(context, attributes);
     }
 
-    attributes.forEach((key, value) -> spanBuilder.setAttribute((AttributeKey) key, value));
+    spanBuilder.setAllAttributes(attributes);
```

Comment:
any ideas how we can init the `SpanBuilder` with the already built `Attributes` map, to avoid this copy?

Reply:
Hmm - I guess it requires an internal use API for that in the SDK. We can try it but it'll be more hacky.

Reply:
just to clarify, did not mean for this pr, something (possibly) for the future..